deafrica/monitoring/s2_gap_filler.py

```diff
@@ -82,7 +109,152 @@ def get_common_message_attributes(stac_doc: Dict) -> Dict:
     return msg_attributes
 
 
-def prepare_message(scene_paths: list, log: Optional[logging.Logger] = None):
+def prepare_s2_l2a_stac(src_stac_doc: Dict):
+    """Prepares the appopriate STAC data to send with the sqs message.
+    The original json/stac_doc must be modified with the appropriate
+    fields. The metadata changes across time which makes this a bit
+    messy for capturing known edge cases.
+
+    Args:
+        src_stac_doc: Original STAC doc/json from gap report
+
+    Returns:
+        stac_metadata: formatted stacmetadata for the sns message
+    """
+
+    # change the properties to align with original sqs message
+    # read in the tileinfo.json file and create a STAC document
+    # using the sentinel_s2_l2a package.
+    # This will form the basis of our SNS message body as it is
+    # most closely aligned with the original message
+    if "tileinfo_metadata" in list(src_stac_doc["assets"].keys()):
+        tileinfo_url = src_stac_doc["assets"]["tileinfo_metadata"]["href"]
+    else:
+        tileinfo_url = src_stac_doc["assets"]["info"]["href"]
+
+    tileinfo = requests.get(tileinfo_url, stream=True)
+    tileinfo = json.loads(tileinfo.text)
+
+    # update the base url to generate STAC metadata for source
+    base_url = f"https://roda.sentinel-hub.com/sentinel-s2-l2a/{tileinfo['path']}"
+    tileinfo_metadata = sentinel_s2_l2a(tileinfo, base_url=base_url)
+
+    # add the links from the tileinfo
+    relevant_links = ["self", "canonical", "derived_from"]
+    links = [x for x in src_stac_doc["links"] if x["rel"] in relevant_links]
+
+    # update the link path to the origin
+    SENTINEL_2_COGS_URL = "https://sentinel-cogs.s3.us-west-2.amazonaws.com"
+    for link in links:
+        link["href"] = link["href"].replace("s3://sentinel-cogs", SENTINEL_2_COGS_URL)
+        if link["rel"] == "derived_from":
+            # add the title and data type
+            link["title"] = "Source STAC Item"
+            link["type"] = "application/json"
+
+    # add links
+    tileinfo_metadata["links"] = links
+
+    # get the proj:shape, href, and proj:transform from tileinfo
```

Comment:
it looks like the opposite - the values from the source stac doc, not from tileinfo? might worth renaming to src_asset_dict for clarity

Reply:
Correct. I have renamed `tileinfo_metadata` to `new_stac_doc` for clarity. We now have a `src_stac_doc` and a `new_stac_doc` which should be clearer.